Code/Framework/AzCore/AzCore/Memory/OSAllocator.cpp

```diff
@@ -14,9 +14,6 @@ namespace AZ
     {
         Create();
         PostCreate();
-#if defined(AZ_ENABLE_TRACING)
-        SetProfilingActive(true);
-#endif
```

Comment:
The code for the non-allocating [Windows INI-style config parser](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Settings/ConfigParser.h#L139-L149) is in development.
We could just have this be set in a settings file that gets defaults this to false.

Reply:
You could use the following logic to read in a profiling enabled setting without hitting dynamic memory.
```c++
        AZ::IO::SystemFileStream configStream("<PathToConfigFile>", AZ::IO::OpenMode::ModeRead);
        bool setProfilingEnabled = false;
        auto parseConfigEntry = [&setProfilingEnabled](const AZ::Settings::ConfigParserSettings::ConfigEntry& configEntry)
        {
            constexpr bool caseSensitiveCmp = false;
            if (AZ::StringFunc::Equal(configEntry.m_keyValuePair.m_key, "profilingenabled", caseSensitiveCmp))
            {
                setProfilingEnabled = true;
            }

            return true;
        };

        AZ::Settings::ConfigParserSettings parserSettings{ parseConfigEntry };

        [[maybe_unused]] auto parseOutcome = AZ::Settings::ParseConfigFile(configStream, parserSettings);
        AZ_Assert(parseOutcome, "Error parsing Config File %s:\n%s", m_configStream.GetFilename(), parseOutcome.error().c_str());
```

Reply:
It should be possible to even load the config file at static initializing time as well.
Such as the first time the allocator is created on demand

Reply:
I like this idea, but I'm going to leave it for a separate change.

Reply:
Future Note: Alternatively, we can even read these a CFG file from the user home directory as well by using the [AZ::Utils::GetHomeDirectory](https://github.com/o3de/o3de/blob/development/Code/Framework/AzCore/AzCore/Utils/Utils.h#L108-L111) method, as that can also be accessed during static initialization.